server/src/main/java/io/crate/execution/engine/collect/CollectTask.java

```diff
@@ -187,7 +187,7 @@ public void addSearcher(int searcherId, RefCountedItem<? extends IndexSearcher>
         synchronized (searchers) {
             if (releasedResources == false) {
                 var replacedSearcher = searchers.put(searcherId, searcher);
-                if (replacedSearcher != null) {
+                if (started.get() && replacedSearcher != null) {
```

Comment:
This is a wrong way to address the issue. Collect task is started right before adding searchers.